### ISSUE TITLE
Implement calculation and pre-determination of insufficient storage status 

### ISSUE THREAD

**Issue by bjester** [learningequality/kolibri#10382]
Depends on: https://github.com/learningequality/kolibri/pull/10365 and https://github.com/learningequality/kolibri/issues/10385

## Summary
With https://github.com/learningequality/kolibri/pull/10365, we implemented the processing of content assigned in lessons and exams in order to generate corresponding sync-initiated download requests for that content. That, in addition to importing the content metadata for those resources, occurs within the cleanup stage of a facility data pull sync. Since the syncable model `LearnerDeviceStatus` tracks a status for whether the device has insufficient storage for learning materials, and the syncing of it allows the coach to have insight into that, being able to determine that status during the pull portion of a sync is worthwhile because the device will then process the 'push' portion of the sync immediately afterwards. This was not implemented to start to allow refinement of our algorithm for determining available storage.

### The function [`synchronize_content_requests`](https://github.com/learningequality/kolibri/blob/c588e4b5d73ae0342d3ff56dd2f4f2bf400e698b/kolibri/core/content/utils/content_request.py#L49)
After importing resource metadata
a) It should query the database to determine the total size of class resources not already downloaded to the device
b) It should query the database to determine the total size of class resources that are no longer needed
c) It should query the database to determine the total size of resources downloaded by the user
d) It should save the `InsufficientStorage` status if the value of (a) - (b) - (c)  >= the amount of allowable content storage (see https://github.com/learningequality/kolibri/issues/10385)

## User-facing consequences
Coaches will have the best insight into learner devices and their storage status if we can immediately sync it back

**Comment by bjester** [comment]
Also add processing that deletes download requests after removals have been processed.
